### PR TITLE
Orca 4578 audit flags

### PR DESCRIPTION
**Pull request by msarvar**
Removed several flags

- All flags related to TFE are removed
- `SkipNoCloneChanges` also removed since we don't rely on that feature
- `DisableRepoLocking` is removed, its purpose was to remove locking from projects. We don't really use that
- Removed bunch of `apply_requirements` flags. Instead of flags we use server side config. 
- Removed `GlobalCfgArgs` struct.
  - Now policy checks are dynamically enabled if `policy_sets` are present in the config
  - To enable platform mode added `globalCfg.EnablePlatformMode()` helper function. 
- Removed `DeleteSourceBranchOnMerge` feature

### CODE REVIEW

File: server/controllers/events/testfixtures/test-repos/simple-yaml/exp-output-autoplan.txt [hunk 1]
```diff
@@ -39,6 +39,7 @@ postplan
 </details>
 Plan: 1 to add, 0 to change, 0 to destroy.
 
+
```
Comment (nishkrishnan) [inline server/controllers/events/testfixtures/test-repos/simple-yaml/exp-output-autoplan.txt:42]: why these changes?

File: server/core/config/parser_validator_test.go [hunk 2]
```diff
@@ -1100,38 +1095,20 @@ workflows:
...
 	r := config.ParserValidator{}
-	globalCfgArgs := valid.GlobalCfgArgs{
-		AllowRepoCfg:  false,
-		MergeableReq:  false,
-		ApprovedReq:   false,
-		UnDivergedReq: false,
-	}
 
-	_, err = r.ParseRepoCfg(tmpDir, valid.NewGlobalCfgFromArgs(globalCfgArgs), "repo_id")
+	_, err = r.ParseRepoCfg(tmpDir, valid.NewGlobalCfg(), "repo_id")
```
Comment (nishkrishnan) [inline server/core/config/parser_validator_test.go:1099]: side note:

would be cool to simplify our config story, we have so many things that just get confusing. RepoConfig, GlobalConfig, UserConfig, lol.  Wonder how we can make this better.

File: server/core/config/valid/global_cfg.go [hunk 3]
```diff
@@ -232,41 +195,47 @@ func NewGlobalCfgFromArgs(args GlobalCfgArgs) GlobalCfg {
...
-		deploymentWorkflow := Workflow{
-			Name:  DefaultWorkflowName,
-			Apply: DefaultApplyStage,
-			Plan:  DefaultPlanStage,
-		}
+func (g GlobalCfg) EnablePolicyChecks() GlobalCfg {
+	repo := &g.Repos[0]
+	repo.ApplyRequirements = append(repo.ApplyRequirements, PoliciesPassedApplyReq)
+	return g
+}
```
Comment (nishkrishnan) [inline server/core/config/valid/global_cfg.go:207]: Would be dope at some point if we could remove this and dynamically support policy checks if the config/policies are present.

File: testing/type_converters.go [hunk 4]
```diff
@@ -1,4 +1,8 @@
-package raw_test
+package testing
+
+// String is a helper routine that allocates a new string value
+// to store v and returns a pointer to it.
+func String(v string) *string { return &v }
```
Comment (nishkrishnan) [inline testing/type_converters.go:5]: Where is this used?  we should be able to use `github.String()` or `aws.String()` for most things.

**Review by nishkrishnan — DISMISSED** [review]

File: server/controllers/events/testfixtures/test-repos/simple-yaml/exp-output-autoplan.txt (hunk 1 above)
Comment (msarvar) [reply server/controllers/events/testfixtures/test-repos/simple-yaml/exp-output-autoplan.txt:42]: in `server/events/templates/planSuccessWrapped.tmpl` I added a newline at the end of the template. Some templates have a new line and some don't. Whenever I encounter templates without a newline I just add it and update e2e templates.

File: server/core/config/parser_validator_test.go (hunk 2 above)
Comment (msarvar) [reply server/core/config/parser_validator_test.go:1099]: yeah, there are too many options to configure your repo.

File: testing/type_converters.go (hunk 4 above)
Comment (msarvar) [reply testing/type_converters.go:5]: Bunch of tests use this. It didn't feel worth to refactor this other than moving this into a testing package.

**Review by nishkrishnan — APPROVED** [review]
